### PR TITLE
docs: improve description of rev requirements for post/put document

### PR DESCRIPTION
**Pull request by mojito317**
## PR summary

The goal of this PR is to improve the description of `_rev`/`rev` requirements for post/put document. I also tried to explain where `_id`/`docId` is necessary similarly to `_rev`/`rev`.

I decided to use `/*...code is here...*/` for code comments and ornamentation lines before and after the explanation plus the commented out codes because with this way maybe the eyes can catch what belongs to what. 

I also fixed the indentation for the examples and removed the unused `documentInfoOptions` variable in `modules/examples/src/main/java/CreateDbAndDoc.java`.

## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [x] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [x] Documentation content changes
- [ ] Other (please describe)

## What is the new behavior?
Explanation in example code how to use `putDocument` along with `postDocument` and:
* `_id` is a MUST at CREATE for `putDocument`
* `_id` is OPTIONAL at CREATE for `postDocument`
* `_id` is generated at CREATE when it is not provided `postDocument`
* `_id` and `_rev` is a MUST at UPDATE
*  `id` parameter is a MUST at UPDATE for `putDocument`
* `rev` parameter can be replaced by `_rev` in the document object both for `postDocument` and `putDocument`
* `id` and `rev` parameter is a MUST at DELETE 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: modules/examples/src/main/java/UpdateDoc.java [hunk 1]
```diff
@@ -53,31 +58,55 @@ public static void main(String[] args) {
...
-                    .postDocument(updateDocumentOptions)
-                    .execute()
-                    .getResult();
+                .postDocument(updateDocumentOptions)
+                .execute()
+                .getResult();
+
+            // ====================================================================
+            // Note: updating the document can also be done with the "putDocument"
+            // method. docId and rev are required for an UPDATE operation:
```
Comment (eiri) [inline modules/examples/src/main/java/UpdateDoc.java:85]:
```suggestion
            // method. docId and rev are required for an UPDATE operation,
```

File: modules/examples/src/main/java/UpdateDoc.java [hunk 2]
```diff
@@ -53,31 +58,55 @@ public static void main(String[] args) {
...
+            //  but rev can be provided in the document object too:
```
Comment (eiri) [inline modules/examples/src/main/java/UpdateDoc.java:86]:
```suggestion
            // but rev can be provided in the document object too:
```

**Review by eiri — APPROVED** [review]
legit.

noting no-op change of `.secrets.baseline`, but `detect-secret` tooling so poor ux, so I don't care if this goes in or get removed.

File: modules/examples/src/main/java/UpdateDoc.java (hunk 1 above)
Comment (mojito317) [reply modules/examples/src/main/java/UpdateDoc.java]: Done in [2e6929d](https://github.com/IBM/cloudant-java-sdk/pull/247/commits/2e6929d1238d0c92191aa2103bc31965d045af49)

File: modules/examples/src/main/java/CreateDbAndDoc.java [hunk 3]
```diff
@@ -14,10 +14,10 @@
 import com.ibm.cloud.cloudant.v1.Cloudant;
 import com.ibm.cloud.cloudant.v1.model.Document;
 import com.ibm.cloud.cloudant.v1.model.DocumentResult;
-import com.ibm.cloud.cloudant.v1.model.GetDocumentOptions;
 import com.ibm.cloud.cloudant.v1.model.Ok;
 import com.ibm.cloud.cloudant.v1.model.PostDocumentOptions;
 import com.ibm.cloud.cloudant.v1.model.PutDatabaseOptions;
+// import com.ibm.cloud.cloudant.v1.model.PutDocumentOptions;
```
Comment (emlaver) [inline modules/examples/src/main/java/CreateDbAndDoc.java:20]: Do we need to add a comment here?
```suggestion
// Uncomment import below if using "putDocument" method
// import com.ibm.cloud.cloudant.v1.model.PutDocumentOptions;
```

**Review by emlaver — APPROVED** [review]

File: modules/examples/src/main/java/CreateDbAndDoc.java (hunk 3 above)
Comment (mojito317) [reply modules/examples/src/main/java/CreateDbAndDoc.java:21]: Done in [61952cb](https://github.com/IBM/cloudant-java-sdk/pull/247/commits/61952cbe80cae1e41a0f5f2d524a668e3c1a6ea1)

**Comment by mojito317** [comment]
Thanks for the reviews! I'm about to merge this!

**Pull request merged**